MetaMask/metamask-extension
# Fix spacing issue in Settings Search input

Fix: https://github.com/MetaMask/metamask-extension/issues/17100

## Explanation
This bug was introduced in https://github.com/MetaMask/metamask-extension/pull/16903 with the use of `.trim()`. As input is sanitized upon entry, always trimming the trailing whitespace will result in the denial of space entry. Using `trimStart()` will produce the desired effect of preventing leading whitespaces, while still allowing for normal space entry in between query terms.

## Screenshots/Screencaps

### Before

(Space key doesn't work)

https://user-images.githubusercontent.com/8112138/211394232-e10d115b-2e12-4b09-9654-de03d9f9af67.mov


### After
https://user-images.githubusercontent.com/8732757/211380347-f4980fb1-9b21-489f-a0d5-ab5eb9c75539.mov

## Manual Testing Steps
1. Click to Settings
2. Attempt to enter a query containing spaces, such as `currency conversion`
3. Ensure that the query can be entered as expected with spaces

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
